Userland/Libraries/LibJS/Runtime/IteratorOperations.cpp

```diff
@@ -69,9 +69,37 @@ Object* iterator_next(Object& iterator, Value value)
 }
 
 // 7.4.6 IteratorClose ( iteratorRecord, completion ), https://tc39.es/ecma262/#sec-iteratorclose
-void iterator_close([[maybe_unused]] Object& iterator)
+void iterator_close(Object& iterator)
 {
-    TODO();
+    auto& vm = iterator.vm();
+    auto& global_object = iterator.global_object();
+
+    // Emulates `completion` behaviour
+    auto* exception = vm.exception();
+    vm.clear_exception();
+    ArmedScopeGuard restore_exception = [&]() {
+        if (exception)
+            vm.set_exception(*exception);
+    };
+
+    auto return_method = iterator.get(vm.names.return_).value_or(js_undefined());
+    if (vm.exception())
+        return;
+    if (return_method.is_nullish())
+        return;
+    if (!return_method.is_function()) {
+        vm.throw_exception<TypeError>(global_object, ErrorType::IterableReturnNotAFunction);
+        return;
+    }
```

Comment:
We have `GetMethod`, so please use it :)

https://github.com/SerenityOS/serenity/blob/5eef07d232fc179f0640bbb8cff575cd239c4d65/Userland/Libraries/LibJS/Runtime/Value.cpp#L1385-L1402

Reply:
Nice! Ill fixup iterator next while im here as well then :)